packages/datagateway-download/src/downloadApiHooks.test.tsx

```diff
@@ -542,4 +739,762 @@ describe('Download Cart API react-query hooks test', () => {
+      // it will just always return the old list.
+
+      let isMutated = false;
+
+      const updatedDownload: Download = {
+        ...mockDownloadItems.find(({ id }) => id === 1),
+        status: 'PREPARING',
+      };
+
+      const updatedFormattedDownload: FormattedDownload = {
+        ...mockFormattedDownloadItems.find(({ id }) => id === 1),
+        status: 'downloadStatus.preparing',
+      };
+
+      axios.get = jest.fn().mockImplementation(() =>
+        Promise.resolve({
+          data: isMutated
+            ? mockDownloadItems.map((download) =>
+                download.id === updatedDownload.id ? updatedDownload : download
+              )
+            : mockDownloadItems,
+        })
+      );
+      axios.put = jest.fn().mockImplementation(() => Promise.resolve());
+
+      const { result, waitFor } = renderHook(
+        () => ({
+          useAdminDownloads: useAdminDownloads({
+            initialQueryOffset: 'LIMIT 0, 50 ',
+          }),
+          useAdminUpdateDownloadStatus: useAdminUpdateDownloadStatus(),
+        }),
+        {
+          wrapper: createReactQueryWrapper(),
+        }
+      );
+
+      await waitFor(() => result.current.useAdminDownloads.isSuccess);
+      isMutated = true;
+      result.current.useAdminUpdateDownloadStatus.mutate({
+        downloadId: 1,
+        status: 'PREPARING',
+      });
+      await waitFor(
+        () => result.current.useAdminUpdateDownloadStatus.isSuccess
+      );
+
+      expect(
+        result.current.useAdminDownloads.data.pages[0].find(
+          ({ id }) => id === 1
+        )
+      ).toEqual(updatedFormattedDownload);
+    });
+
+    it('should call handleICATError and rollback optimistic changes if an error is encountered', async () => {
+      axios.put = jest.fn().mockRejectedValue({
+        message: 'Test error message',
+      });
+      axios.get = jest.fn().mockResolvedValue({ data: mockDownloadItems });
+
+      const { result, waitFor } = renderHook(
+        () => ({
+          useAdminDownloads: useAdminDownloads({
+            initialQueryOffset: 'LIMIT 0, 50',
+          }),
+          useAdminUpdateDownloadStatus: useAdminUpdateDownloadStatus(),
+        }),
+        { wrapper: createReactQueryWrapper() }
+      );
+
+      await waitFor(() => result.current.useAdminDownloads.isSuccess);
+      result.current.useAdminUpdateDownloadStatus.mutate({
+        downloadId: 1,
+        status: 'PREPARING',
+      });
+      await waitFor(() => result.current.useAdminUpdateDownloadStatus.isError);
+
+      expect(handleICATError).toHaveBeenCalledWith({
+        message: 'Test error message',
+      });
+      expect(result.current.useAdminDownloads.data.pages).toEqual([
+        mockFormattedDownloadItems,
+      ]);
+    });
+  });
+
+  describe('useSubmitCart', () => {
+    it('should submit cart and clear cart on success', () => {
+      axios.put = jest.fn().mockResolvedValue(123);
+    });
+  });
```

Comment:
Seems there's a test missing here? We mock `axios` but there's no call of `useSubmitCart` or any assertations

Reply:
Ahaha yeah I missed it, sorry about that.